docs/auth/github-api.md

```diff
@@ -0,0 +1,25 @@
+# GitHub API Access
+
+## Required Information
+Since we're using a GitHub App, it's required to store the following persistent information:
+* GitHub App Name - Used in HTTP Requests in the `User-Agent` header
+* GitHub App Id - Used in the JWT, which is used to request an installation token
+* GitHub App Client Id - Used while requesting an OAuth access token   
+* GitHub App Client Secret - Used while requesting an OAuth access token
+* GitHub App Private Key - Used to sign the JWT, which is used to request an installation token
+
+We also need to store the following information:
+* GitHub User Access Token - Used to validate each user's identity and their corresponding information
+* GitHub App Installations
+    * GitHub App Installation Id - Used to identify a GitHub App Installation in an organization
+    * GitHub App Installation Access Token - Used to execute actions in an installation
+    * Github App Installation Access Token Expiration Date - Used to determine when to request a new Installation Access Token
+
+## Obtaining the Information
+The persistent information described above is obtained during the GitHub App's creation. Some of this information should remain private, most notably the app's private key and client secret. This information should also be easily modifiable since the GitHub App can change at any time.
+
+### Installation Access Token
+To obtain an installation access token, the API will have an endpoint which will respond with the GitHub App installation URI, which will be used to install the GitHub App to an organization. The browser will then present the installation form to the user and, after the user picks and installs the app into an organization, the browser will be redirected to a new URI, where the installation ID is present in its query string. This endpoint is part of what we call **CodeGarten Interaction Manager (IM)**, which is neither the API nor a client app. Its only responsibility is to respond to requests such as GitHub Installations and GitHub/CodeGarten Authentications. Via this endpoint, the API will be able to receive the installation ID, and store this information in the database alongside the organization's ID. This interaction should be made on a new browser window, which will close itself in the end.

```

Comment:
I don't understand the "which will respond with the GitHub App installation URI".